securesystemslib/metadata.py

```diff
@@ -94,3 +89,72 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+    def sign(self, signer: Signer) -> Signature:
+        """Sign the payload and create the signature.
+
+        Arguments:
+            signer: A "Signer" class instance.
+
+        Returns:
+            A "Signature" instance.
+        """
+
+        signature = signer.sign(self.pae())
+        self.signatures.append(signature)
+
+        return signature
+
+    def verify(self, keys: List[Key], threshold: int) -> Set[Key]:
+        """Verify the payload with the provided Keys.
+
+        Arguments:
+            keys: A list of "Key" class instance that is a public key to verify
+                the signatures.
+            threshold: Number of signatures needed to pass the verification.
+
+        Raises:
+            ValueError: If "threshold" is not valid.
+            SignatureVerificationError: If the amount of "recognized_signers" is less
+                than provided threshold.
+
+        Returns:
+            recognized_signers: A set of recognized public keys.
+        """
+
+        recognized_signers = set()
```

Comment:
This is a bit tricky. In the case of `SSlibKey.__hash__` the "opaque" key content is now required to be hashable, which is probably okay, but it makes it less "opaque". And in the case of `GPGKey.__hash__`, you are using non-canonical dict/json, which is not guaranteed to always yield the same data for the same key. 

I suggest to use `keyid` as unique identifier.

Reply:
I am not sure about `keyid` at this point because its somewhere between optional and required fields..

The other solution to hashable is, we can create a container for ACCEPTED_KEYS (`KeyList`). which will have an `add` method and we can write an optimized algorithm to add unique keys to it.

Reply:
I really think `keyid` should be mandatory for `Key` objects. See https://github.com/in-toto/securesystemslib/pull/2#discussion_r904006951.

Reply:
More details about why keyids should be mandatory in a `Key` object: 
https://github.com/secure-systems-lab/securesystemslib/issues/310#issuecomment-1191262080